frontend/ui/data/keyboardlayouts/en_keyboard.lua

```diff
@@ -55,73 +57,151 @@ local _Y_ = en_popup._Y_
+            { _2p, _2_, "‘", "¡", },
+            { _3p, _3_, "’", dsh, },
+            { _4p, _4_, "“", "_", },
+            { _5p, _5_, "”", quo, },
+            { _6p, _6_, eq2, eql, },
+            { _7p, _7_, _7s, _7n, },
+            { _8p, _8_, _8s, _8n, },
+            { _9p, _9_, _9s, _9n, },
+            { _0p, _0_, mn2, mns, },
         },
-        -- second row
-        {  --  1       2       3       4       5       6       7       8
-            { _A_,    _a_,    "…",    _at,    "Ê",    "ê",    "Ş",    "ş", },
-            { _S_,    _s_,    "$",    "4",    "Ë",    "ë",    "İ",    "ı", },
-            { _D_,    _d_,    "%",    "5",    "Î",    "î",    "Ğ",    "ğ", },
-            { _F_,    _f_,    "^",    "6",    "Ï",    "ï",    "Ć",    "ć", },
-            { _G_,    _g_,    ":",    ";",    "Ô",    "ô",    "Č",    "č", },
-            { _H_,    _h_,    '"',    "'",    "Œ",    "œ",    "Đ",    "đ", },
-            { _J_,    _j_,    "{",    "[",    "Ù",    "ù",    "Š",    "š", },
-            { _K_,    _k_,    "}",    "]",    "Û",    "û",    "Ž",    "ž", },
-            { _L_,    _l_,    "_",    "-",    "Ÿ",    "ÿ",    "Ő",    "ő", },
+        -- Second row
+        { --   R    r    S    s
+            { _Q_, _q_, dif, "?", },
+            { _W_, _w_, int, "¿", },
+            { _E_, _e_, mth, "~", },
+            { _R_, _r_, mt2, "\\", },
+            { _T_, _t_, df2, bar, },
+            { _Y_, _y_, sl2, sla, },
+            { _U_, _u_, _4s, _4n, },
+            { _I_, _i_, _5s, _5n, },
+            { _O_, _o_, _6s, _6n, },
+            { _P_, _p_, pl2, pls, },
         },
-        -- third row
-        {  --  1       2       3       4       5       6       7       8
-            { "",   "",   "2/2",  "1/2",   "",   "",   "",    "",
-              width = 1.5
-            },
-            { _Z_,    _z_,    "&",    "7",    "Á",    "á",    "Ű",    "ű", },
-            { _X_,    _x_,    "*",    "8",    "Ø",    "ø",    "Ã",    "ã", },
-            { _C_,    _c_,    "£",    "9",    "Í",    "í",    "Þ",    "þ", },
-            { _V_,    _v_,    "<",    com,    "Ñ",    "ñ",    "Ý",    "ý", },
-            { _B_,    _b_,    ">",    prd,    "Ó",    "ó",    "†",    "‡", },
-            { _N_,    _n_,    "‘",    "↑",    "Ú",    "ú",    "–",    "—", },
-            { _M_,    _m_,    "’",    "↓",    "Ç",    "ç",    "…",    "¨", },
+        -- Third row
+        { --   R    r    S    s
+            { _A_, _a_, ls2, ls1, },
+            { _S_, _s_, mr2, mr1, },
+            { _D_, _d_, dgr, "(", },
+            { _F_, _f_, tpg, ")", },
+            { _G_, _g_, hs2, hsh, },
+            { _H_, _h_, pd2, pdc, },
+            { _J_, _j_, _1s, _1n, },
+            { _K_, _k_, _2s, _2n, },
+            { _L_, _l_, _3s, _3n, },
+            { sec, cse, sec, cse, }, -- comma/semicolon with CSS popup block
+        },
+        -- Fourth row
+        { --   R    r    S    s
+            { label = "",
+              width = 1.5, },           -- Shift
```

Comment:
You can make these a single line for better readability.

Reply:
OK!